modules/ocf_irc/manifests/xmpp.pp

```diff
@@ -0,0 +1,60 @@
+class ocf_irc::xmpp {
+  # lua-dbi-mysql is needed to connect to MySQL
+  #package { ['lua-dbi-mysql']: }
+
+  # When upgrading to buster, these can be installed from the regular buster
+  # repo
+  ocf::repackage {
+    'lua-dbi-common':
+      backport_on => ['stretch'];
+
+    'lua-dbi-mysql':
+      backport_on => ['stretch'];
+
+    'prosody':
+      backport_on => ['stretch'];
+
+    'prosody-modules':
+      backport_on => ['stretch'];
+  }
+
+  service { 'prosody':
+    enable    => true,
+    require   => Package['prosody'],
+    subscribe => Ocf::Ssl::Bundle[$::fqdn],
+  }
+
+  # Make the prosody user able to read the certs
+  user { 'prosody':
+    groups  => 'ssl-cert',
+    require => [Package['prosody'], Package['ssl-cert']],
+  }
+
+  ocf::firewall::firewall46 {
+    '101 allow xmpp':
+      opts   => {
+        chain  => 'PUPPET-INPUT',
+        proto  => 'tcp',
+        dport  => [5222, 5269],
+        action => 'accept',
+      }
+  }
+
+  $vhost_name = $::host_env ? {
+    'dev'  => 'dev-xmpp.ocf.berkeley.edu',
+    'prod' => 'ocf.berkeley.edu',
```

Comment:
Perhaps document that we are using a special SRV record so this is correct as is?